### ISSUE TITLE
Basic ML Implementations (Any Topic)

### ISSUE THREAD

**Issue by abhinand5** [abhinand5/ml-notebooks-101#1]
# Implementation of Basic ML stuff (from scratch or without extensive use of higher-level functions)

## Description  
If you think you can write notebook/scripts that can vastly help beginners understand the ins outs of the chosen ML algorithm/problem. Please go ahead and reply in this thread. 

## Instructions
- Use Python
- Don't make a pull request unless you are assigned with the task. 
- Add your file in an appropriate folder with README.md in it. Add requirements.txt if needed.
- Exclude data files or saved model files if they are more than 20MB but include the scripts to automatically download or point to a source.
- Use proper coding guidelines 

> **Note: If you are making changes to existing code, you're welcome, but clearly mention these things**
* What changes have been made?
* Why the changes have been made? 
* How is the new change better? 

### In Progress:
- [x] Intro to ML with Titanic Dataset - @vibhorkrishna
- [x] Basic ANN implementation for Housing Price Prediction - @Gunnika
- [x] Text Classification using LSTM & GloVe -  @arunpandian7

**Comment by vibhorkrishna** [comment]
How much basic implementation do you want. Are we allowed to use libraries like label encoder?
You can assign this task to me.
I will work on very basic and famous dataset of Titanic: https://www.kaggle.com/c/titanic
I will also put comments on my code so that the code can be easily understood by the beginners

**Comment by abhinand5** [comment]
@vibhorkrishna 
Yeah sure go ahead. You can try to implement things in a way to help beginners get a solid understanding of the underlying stuff. For data manipulation and related you can of course use sklearn utilities.

**Comment by vibhorkrishna** [comment]
Hey man I was unable to send the pull request through this issue. An another issue was created. Please label that issue as hacktoberfest and hacktoberfest-accepted.

**Comment by vibhorkrishna** [comment]
Can you tell me what is label checker that is saying my checks were not passed?
And how do I correct it?

**Comment by abhinand5** [comment]
Don't worry about that workflow, I've removed it as I didn't find it useful.

**Comment by Gunnika** [comment]
Hi
I've added a Pull Request for predicting house price with ANN. #3 
Also, for repositories to be eligible for Hacktoberfest, they need to have the 'hacktoberfest' label. 
Thanks

**Comment by abhinand5** [comment]
Hi @Gunnika, thanks I have added the `hacktoberfest` tag as well. Please go through the PR thread #3  and do the suggested changes.

**Comment by arunpandian7** [comment]
Hi @abhinand5 , I would love to add **Text Classification using LSTM & GloVe** notebook. Can you assign it to me and I make my PR?

**Comment by abhinand5** [comment]
Sure go ahead @arunpandian7

**Comment by AdiNarendra98** [comment]
Hi,
I would like to work on contributing to making python implementation notebooks for beginners in some ML Algorithms.These shall act as a their first Intro guide for "ML Using Python". 
Ex- First Python Implementation of Linear Regression or First Python Implementation of Logistic Regression. 
Can you assign it to me ?
Thank in advance.

**Comment by SapnilPatel** [comment]
Hello! I want to contribute to the Implementation of GAN. So, Can you please assign me a task?

**Comment by aurvl** [comment]
Hello! I would like to contribute to this repository by implementing a predictive model for time series data. My goal is to compare time series-based models like LSTM and SARIMA, which rely on time-indexed data, with ensemble models like Random Forest or XGBoost, which require features for prediction. The idea is to demonstrate how the former models leverage the temporal nature of the data, while models like RF need the data to be transformed into explicit features for prediction. Could you please assign me a task and explain the steps I should follow to get started? This will be my first contribution!

**Comment by abhinayaanair** [comment]
Can we use libraries like Scikit-learn for vectorization or model training?

I’d like to take on this task.
For this, I’ll work on a classic Spam Classification problem using the SMS Spam Collection dataset (available at: https://www.kaggle.com/datasets/uciml/sms-spam-collection-dataset.

I’ll preprocess the text data by cleaning and tokenizing it.
For feature extraction, I’ll implement Bag of Words and optionally TF-IDF using Scikit-learn.
I’ll train a Naive Bayes classifier (MultinomialNB) to distinguish between spam and non-spam messages.
I’ll evaluate the model using accuracy, precision, recall, and F1-score to ensure beginners understand the importance of performance metrics.

**Comment by Divyaanshvats** [comment]
@abhinand5  Is this project still open, I am interested in joining especially working with Deep Learning models?

**Comment by vansh-virmani** [comment]
I want to work on Logistic regression implementation from scratch using python using sigmoid fucntion
